src/aws/osml/image_processing/sar_complex_imageop.py

```diff
@@ -0,0 +1,157 @@
+import logging
+from typing import Optional, Tuple
+
+import numpy as np
+
+TWO_PI = np.pi * 2.0
+
+logger = logging.getLogger(__name__)
+
+
+def image_pixels_to_complex(
+    image_pixels: np.ndarray, pixel_type: Optional[str] = None, amplitude_table: Optional[np.typing.ArrayLike] = None
+) -> np.ndarray:
+    """
+    This function converts SAR pixels from SICD imagery into complex values using equations
+    found in SICD Volume 1 Section 4.2.
+
+    :param image_pixels: the SAR image pixels
+    :param pixel_type: "AMP8I_PHS8I", "RE32F_IM32F", or "RE16I_IM16I"
+    :param amplitude_table: optional lookup table of amplitude values for AMP8I_PHS8I image pixels
+    :return:
+    """
+
+    if pixel_type is None or pixel_type in ["RE32F_IM32F", "RE16I_IM16I"]:
+        # For these pixel types the complex value is already stored in the file
+        return image_pixels
+    elif pixel_type == "AMP8I_PHS8I":
+        # If the data is 8-bit amplitude/phase with an optional amplitude lookup table need to
+        # convert it to the complex image value
+        amplitude = image_pixels[0]
+        phase = image_pixels[1] / 256.0
+        if amplitude_table is not None:
+            amplitude_lut = np.array(amplitude_table)
+            amplitude = amplitude_lut[amplitude]
+        return np.array([amplitude * np.cos(TWO_PI * phase), amplitude * np.sin(TWO_PI * phase)])
+    else:
+        raise ValueError(f"Unknown SAR Pixel Type: {pixel_type}")
+
+
+def complex_to_power_value(complex_data: np.ndarray) -> np.ndarray:
+    """
+    This function converts SAR complex data into the pixel power values (sometimes
+    called pixel intensity) using the equation found in SICD Volume 1 Section 4.10.
+
+    :param complex_data: the SAR complex image signal with real and imaginary components
+    :return: the power values
+    """
+    return np.sum(np.square(complex_data), axis=0)
+
+
+def power_value_in_decibels(power_values: np.ndarray) -> np.ndarray:
```

Comment:
Do we want to write unit tests for these functions?

Reply:
I'd like to but I don't have a good set of values for them yet. I need to do some more digging otherwise I'll just end up running some numbers through and guessing that whatever came out the other end is correct.